deltacat/compute/compactor/steps/materialize.py

```diff
@@ -41,6 +41,54 @@ def materialize(
         schema: Optional[pa.Schema] = None,
         deltacat_storage=unimplemented_deltacat_storage) -> MaterializeResult:
 
+    def _materialize(
```

Comment:
why not define this in global namespace?

Reply:
Mainly for readability and convenience. This function is specific to the materialize step, and moving this out to the global namespace means more variables to pass around.

Reply:
Makes sense..